### PR TITLE
Azure logging stack 

### PR DESCRIPTION
**Pull request by aditya-facets**
# Description 
Added helm chart for azure logging substack 
- Fluentd 
- Wetty 
- Wetty-ssh 
- wetty services 
- Support to mount additional pvc (logs-pvc) [Default : false ]

# Testing 
- When additional mounting is disabled 
``` 
wettysshAdditionalVolumes:
  enabled: false
```
PVC
image
Pods status 
image

- When additional mounting is enabled 
``` 
wettysshAdditionalVolumes:
  enabled: true
```
-  event from wetty-ssh pod when logs-pvc is not available to be additionally mounted
     
image

> When logs pvc was added additionaly to be mounted 

   
image
image
       

> Pod status 

image
      

> Services for wetty and wetty-ssh 

image

### CODE REVIEW

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      port: 22
+      protocol: TCP
+      targetPort: 22
+  selector:
+    app: wetty-ssh
+    release: wetty-ssh
+  type: ClusterIP
+
+---
+apiVersion: v1
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:122]: skip this secret and all references to it

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            readOnly: true
+          {{- end }}
+          {{- end }}
+      tolerations:
+      - effect: NoSchedule
+        key: kubernetes.azure.com/scalesetpriority
+        operator: Equal
+        value: spot
+      volumes:
+      - hostPath:
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:61]: Remove this and corresponding mount

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+          path: /var/log/dumps/wetty-ssh
+          type: DirectoryOrCreate
+        name: dump-vol
+      - name: wetty-ssh-file-credentials-vol
+        secret:
+          defaultMode: 420
+          secretName: wetty-ssh-file-credentials
+      - name: logs-vol-alpha
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:69]: Rename to logs-vol-azure-files

File: azure-logging-stack/templates/storage.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.persistentVolumeClaims.name }}-pvc
+spec:
+  accessModes:
+    - {{ .Values.persistentVolumeClaims.accessMode }}
+  storageClassName: {{ .Values.storageClassName }}
+  resources:
+    requests:
+      storage: {{ .Values.persistentVolumeClaims.storageSize }}
```
Comment (rr0hit) [inline azure-logging-stack/templates/storage.yaml:31]: check if you increase this value is the volume autoexpanding or not

File: azure-logging-stack/values.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+chartName: loggingstack
```
Comment (rr0hit) [inline azure-logging-stack/values.yaml:1]: Organize the values properly. Users should be able to understand the structure. 

gloabal: (if any)
wetty:
wetty-ssh:
fluentd:
logs-volume:

File: azure-logging-stack/values.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+
+persistentVolumeClaims:
+  name: logs-nfs
+  accessMode: ReadWriteMany
+  storageSize: 10Gi
+
+
+
+priorityClass:
```
Comment (rr0hit) [inline azure-logging-stack/values.yaml:16]: Take this as 

fluentd: 
  priorityClassName: <>

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+          imagePullPolicy: IfNotPresent
+          name: wetty-ssh
+          ports:
+          - containerPort: 22
+            name: ssh
+            protocol: TCP
+          resources:
+            limits:
+              cpu: "2"
+              memory: 4Gi
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:37]: Take these values as paramenters. Keep defaults low. 0.5core and 1GB

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+          volumeMounts:
+          - mountPath: /var/log/dumps
+            name: dump-vol
+          - mountPath: /etc/credentialfiles/
+            name: wetty-ssh-file-credentials-vol
+          - mountPath: /mnt/data
+            name: data-vol
+          - mountPath: /var/log/alpha/efs
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:45]: let the logs path be /var/log/efs. If they mount existing pvc let them provide a different mount path

**Review by rr0hit — CHANGES_REQUESTED** [review]

File: azure-logging-stack/templates/storage.yaml (hunk 4 above)
Comment (aditya-facets) [reply azure-logging-stack/templates/storage.yaml:31]: # Testing 

**Volume expansion** 

- `storageSize: 15Gi`

image

- `storageSize: 20Gi`

image

File: azure-logging-stack/templates/fluentd.yaml [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  namespace: default
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Values.global.chartName }}-fluentd
+      release: fluentd
+  template:
+    metadata:
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
```
Comment (rr0hit) [inline azure-logging-stack/templates/fluentd.yaml:18]: remove this annotation

File: azure-logging-stack/templates/storage.yaml
Comment (rr0hit) [inline azure-logging-stack/templates/storage.yaml:1]: delete the file

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 10]
```diff
@@ -0,0 +1,136 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  annotations:
+    cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:5]: remove this annotation

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+  selector:
+    matchLabels:
+      app: {{ .Values.global.chartName }}-wetty-ssh
+      release: wetty-ssh
+  serviceName: wetty-ssh
+  template:
+    metadata:
+      annotations:
+        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
+        sidecar.istio.io/inject: "true"
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:21]: remove these annotations

File: azure-logging-stack/values.yaml [hunk 12]
```diff
@@ -0,0 +1,75 @@
+global:
+  chartName: loggingstack
```
Comment (rr0hit) [inline azure-logging-stack/values.yaml:2]: why take this? Just use Release.Name that is available by default

File: azure-logging-stack/templates/fluentd.yaml [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+---
+# PVC
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.logsVolume.persistentVolumeClaims.name }}-pvc
+spec:
+  accessModes:
+    - {{ .Values.logsVolume.persistentVolumeClaims.accessMode }}
+  storageClassName: {{ .Values.logsVolume.storageClassName }}
```
Comment (rr0hit) [inline azure-logging-stack/templates/fluentd.yaml:60]: keep this fixed to the azure files one

File: azure-logging-stack/templates/fluentd.yaml [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+---
+# PVC
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.logsVolume.persistentVolumeClaims.name }}-pvc
+spec:
+  accessModes:
+    - {{ .Values.logsVolume.persistentVolumeClaims.accessMode }}
```
Comment (rr0hit) [inline azure-logging-stack/templates/fluentd.yaml:59]: always readwritemany as fluentd pods will use this

File: azure-logging-stack/values.yaml [hunk 15]
```diff
@@ -0,0 +1,74 @@
+global:
+  tolerations:
+    effect: NoSchedule
+    key: kubernetes.azure.com/scalesetpriority
+    operator: Equal
+    value: spot
+
+logsVolume:
+  storageClassName: ccazfile
+  persistentVolumeClaims:
```
Comment (rr0hit) [inline azure-logging-stack/values.yaml:10]: make persistentVolumeClaims -> persistentVolumeClaim

File: azure-logging-stack/templates/fluentd.yaml [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+              name: logs-vol
+      priorityClassName: {{ .Values.fluentd.priorityClass.name }}
+      restartPolicy: {{ .Values.fluentd.restartPolicy }}
+      tolerations:
+        - effect: {{ .Values.global.tolerations.effect }}
+          key: {{ .Values.global.tolerations.key }}
+          operator: {{ .Values.global.tolerations.operator }}
+          value: {{ .Values.global.tolerations.value }}
+      volumes:
+        - name: logs-vol
```
Comment (rr0hit) [inline azure-logging-stack/templates/fluentd.yaml:42]: lets have the name here also set to logs-vol-azure-files

File: azure-logging-stack/Chart.yaml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+name: azure-logging-stack
```
Comment (rr0hit) [inline azure-logging-stack/Chart.yaml:2]: azure-nfs-log-collector

File: azure-logging-stack/templates/fluentd.yaml [hunk 18]
```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: {{ .Release.Name }}-fluentd
+  name: {{ .Release.Name }}-fluentd
+  namespace: default
```
Comment (rr0hit) [inline azure-logging-stack/templates/fluentd.yaml:7]: dont hardcode namepace

File: azure-logging-stack/values.yaml [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+    storageSize: 20Gi
+
+wettySsh:
+  name: wetty-ssh
+  image: 313496281593.dkr.ecr.us-east-1.amazonaws.com/wetty-ssh-host:30a4b00-16
+  resources:
+    limits:
+      cpu: "0.5"
+      memory: 1Gi
+  wettysshAdditionalVolumes:
```
Comment (rr0hit) [inline azure-logging-stack/values.yaml:19]: just have
wettysshAdditionalVolumes:
      - name: logs-vol
        mountPath: /var/log/efs
        claimName: logs-pvc
     - name: my2nd
       mountPath: /var/log/efs
       claimName: logs-pvc2

File: azure-logging-stack/templates/wetty-ssh.yaml [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+    - apiVersion: v1
+      kind: PersistentVolumeClaim
+      metadata:
+        name: data-vol
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 5Gi
```
Comment (rr0hit) [inline azure-logging-stack/templates/wetty-ssh.yaml:76]: make this configurable

**Review by rr0hit — CHANGES_REQUESTED** [review]

File: azure-nfs-log-collector/templates/fluentd.yaml [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+---
+# PVC
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Release.Name }}-pvc
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: ccazfile
```
Comment (rr0hit) [inline azure-nfs-log-collector/templates/fluentd.yaml:60]: Let's take this as a helm param afterall.

**Review by rr0hit — CHANGES_REQUESTED** [review]

**Review by rr0hit — CHANGES_REQUESTED** [review]

**Pull request merged**